### DISCUSSION TITLE
ImageMosaic plugin no longer available

### DISCUSSION THREAD

**Discussion by fabio1299** [kartoza/docker-geoserver#544] [Q&A] [answered]
I'm relatively new to Geoserver and I've inherited a Geoserver installation that relies on the kartoza docker image 2.17.2.
I'm trying to upgrade it to the latest Geoserver (or at least a newer version).
The problem I have is that the current (2.17.2) installation loads the ImageMosaic extension, which does not seem to available if I use a newer version of the kartoza docker images (tried 2.23.1 and 2.21.4).
Both return similar errors:

> geoserver_1  | Plugin URL does not exist::  https://sourceforge.net/projects/geoserver/files/GeoServer/2.23.1/extensions/geoserver-2.23.1-imagemosaic-jdbc-plugin.zip
> geoserver_1  |  imagemosaic-jdbc-plugin extension will not be installed because it is not available

> geoserver_1  | Plugin URL does not exist::  /2.21.4/extensions/geoserver-2.21.4-imagemosaic-jdbc-plugin.zip
> geoserver_1  |  imagemosaic-jdbc-plugin extension will not be installed because it is not available

Thanks for your advice.

**Comment by NyakudyaA** [comment]
I think the url where the plugins are fetched might have changed. Can you try to set an env variable for the

STABLE_PLUGIN_BASE_URL. Can you open an issue for this and I will fix

**Reply by NyakudyaA** [reply]
Please take a look here, https://sourceforge.net/projects/geoserver/files/GeoServer/2.23.1/extensions/ it seems the plugin no longer exists for this specific version of GeoServer. Please consult the changelog for each successive version

**Reply by fabio1299** [reply]
Thank for your answer, I had already checked the link you suggest.

The plugin still exists up to Geoserver 2.23, see this here, https://docs.geoserver.org/stable/en/user/data/raster/imagemosaic/index.html

Any suggestions on why wouldn't it have made it into the URL you indicate?

Thanks

**Reply by NyakudyaA** [reply, marked as answer]
I think you need to ask the GeoServer user group because they would know better
